fix(content): "Remnant: Continue Research" no longer offers if you've completed the missions after it (#10056)
endless-sky/endless-sky: data/human/deep missions.txt

	to offer
		has "event: deep: remnant: rest period"
		has "deep: did reveal remnant"

> 
> # If the player finished the following missions prior to the addition of this one in v0.9.15, it will offer and lead to nothing.
> 

Reviewer: ```suggestion
		# If the player received subsequent missions prior to the addition of this one in v0.9.15, do not offer this one.
```

> 		not "Deep: Remnant: Engines: offer"

Reviewer 2: ```suggestion
		not "Deep: Remnant: Engines: offered"
```
_so close_

Author: Oops.
